cg/store/crud/create.py

```diff
@@ -398,16 +398,18 @@ def link_case_to_order(order_id: int, case_id: int):
 
     def add_illumina_flow_cell(self, flow_cell_dto: IlluminaFlowCellDTO) -> IlluminaFlowCell:
         """Add a new Illumina flow cell to the status database as a pending transaction."""
-        if self.get_illumina_flow_cell_by_internal_id(flow_cell_dto.internal_id):
-            raise ValueError(f"Flow cell with {flow_cell_dto.internal_id} already exists.")
-        new_flow_cell = IlluminaFlowCell(
-            internal_id=flow_cell_dto.internal_id,
-            type=flow_cell_dto.type,
-            model=flow_cell_dto.model,
-        )
-        self.session.add(new_flow_cell)
-        LOG.debug(f"Flow cell added to status db: {new_flow_cell.internal_id}.")
-        return new_flow_cell
+        try:
+            self.get_illumina_flow_cell_by_internal_id(flow_cell_dto.internal_id)
+        except ValueError:
+            new_flow_cell = IlluminaFlowCell(
+                internal_id=flow_cell_dto.internal_id,
+                type=flow_cell_dto.type,
+                model=flow_cell_dto.model,
+            )
+            self.session.add(new_flow_cell)
+            LOG.debug(f"Flow cell added to status db: {new_flow_cell.internal_id}.")
+            return new_flow_cell
+        raise ValueError(f"Flow cell already exists: {flow_cell_dto.internal_id}")
```

Comment:
I think it is better to raise a custom exception here, like `FlowCellNotFound` or something similar

Reply:
I added an "EntryNotFoundError" and "EntryAlreadyExistsError" where applicable.